### DISCUSSION TITLE
[Feature Request] per-row hold Held/LetGo/Missed scores in upload xml and theme

### DISCUSSION THREAD

**Discussion by teleshoes** [TeamRizu/OutFox#244] [03 - Feature Requests]
in the Save/Upload/*.xml files, `<HoldNoteScores>` note counts count separate notes in jumps independently (unlike tap timing windows, which count jumps per-row will count a jump as a single `Fantastic` or `Perfect` or `Great`).

i would like a config option to change this behavior, OR to always record both. (this may not be trivial to implement, as the judgment needs to keep in mind whether the hold started at the same time as another hold in the past, so to speak)

(note that there is a config option to count jumps as a single combo, instead of increasing the combo by the number of notes. this is not that feature)

e.g.:
- lets say you have a song:
    ![holds-5-or-3](https://user-images.githubusercontent.com/592062/97820236-345d0a80-1c7b-11eb-97f4-0d63ae12270a.png)

- in the score screen in soundwaves, you will have:
    **ACTUAL**: `Holds 5/5`
    **DESIRED**: `Holds 3/3`

- in the upload XML file, you will have:
    **ACTUAL**
    ```
    <HoldNoteScores>
    <LetGo>0</LetGo>
    <Held>5</Held>
    <MissedHold>0</MissedHold>
    </HoldNoteScores>
    ```
    **DESIRED**
    ```
    <HoldNoteScores>
    <LetGo>0</LetGo>
    <Held>3</Held>
    <MissedHold>0</MissedHold>
    </HoldNoteScores>
    ```
    or just always do:
    ```
    <HoldNoteScores>
      <LetGo>0</LetGo>
      <Held>5</Held>
      <MissedHold>0</MissedHold>
    </HoldNoteScores>
    <HoldNotePerRowScores>
      <LetGoPerRow>0</LetGoPerRow>
      <HeldPerRow>3</HeldPerRow>
      <MissedHoldPerRow>0</MissedHoldPerRow>
    </HoldNotePerRowScores>
    ```

**Comment by bonimy** [comment]
I second this. It is currently impossible to correctly emulate DDR A scoring without this feature.

**Reply by teleshoes** [reply]
very true, but you can approximate it pretty well by multiplying the ratio of OK arrows to total **hold arrows** by the number of **hold rows**.

below is the tool that i use to generate and track my scores in `DDR A` format:
https://github.com/teleshoes/ddr-stepmania-tools/blob/master/src/ddr-stepmania-score#L221

note that it relies on other tools in the same repo, like `simfile-radar` to count the number of DDR-style hold rows in the SM/SSC stepchart file. it maybe only works on linux, i dunno, and is generally not very user friendly, but if you wanted to do something like this, it could be a good jumping-off place. i have a fairly large corpus of data for validation and testing.

estimating the hold rows in terms of the hold arrows has some advantages:
- exactly correct when you hit all holds
- exactly correct when you miss all holds
- stable + fair (changing any arrow from MISS to OK cannot decrease your ddr score, and OK=>MISS cannot increase it)
- corresponds well with skill level, although not necessarily proportionally

of course, in certain edge cases (e.g.: missing just one arrow in a lot of jumps), the approximate ddr score can be wildly different.
